### PR TITLE
add support for TrackEvent and Experiment

### PR DESCRIPTION
**Pull request by jkaho**
- bump to v4.5.1
- also includes some Promotions changes

### CODE REVIEW

File: src/generated/api/eventsApi.ts [hunk 1]
```diff
@@ -112,11 +112,13 @@ export class EventsApi {
...
    * @summary Send event
+   * @param accountId Unlike other API calls, the &#x60;SendEvent&#x60; call requires the &#x60;Account-Id&#x60; header to be set to your account ID. This is because other API calls infer the account ID from your API key.
    * @param sendEventRequest
    */
   public async sendEvent(
+    accountId: string,
```
Comment (jonathaningram) [inline src/generated/api/eventsApi.ts:121]: Unfortunately this is a major version bump:

> MAJOR version when you make incompatible API changes,

https://semver.org/

That's OK though, this is our reality. We just go to 6.0.0.

**Review by jonathaningram — CHANGES_REQUESTED** [review]

**Comment by jonathaningram** [comment]
We also wrap this (for better or worse) and we did for SendEvent. So if you follow that trail, we should do the same for TrackEvent, though we can probably just put this in collections.ts not events.ts. https://github.com/sajari/sdk-node/blob/master/src/events.ts

File: src/generated/api/eventsApi.ts (hunk 1 above)
Comment (jonathaningram) [reply src/generated/api/eventsApi.ts:121]: Ah if we're at v4 now, then yep v5. I thought we were on v5 already but I must have misread.

And yep, if the account ID field is required for an RPC, we should pass accountId as the first argument in those cases. If it's optional, I suppose we're going to need to allow them to pass it another way?

**Review by jonathaningram — CHANGES_REQUESTED** [review]
commented

**Comment by jkaho** [comment]
@jonathaningram I assume this doesn't call for a major version bump anymore?

**Comment by jonathaningram** [comment]
@jkaho unfortunately it's still a breaking change. E.g. the signature for this is now:

```
public async sendEvent(
    sendEventRequest: SendEventRequest,
    accountId?: string,
    options: { headers: { [name: string]: string } } = { headers: {} }
  ): Promise<{ response: http.IncomingMessage; body: object }>
```

Before users could call it like `sendEvent(req, opts)` now they have to call it like `sendEvent(req, undefined, opts)`. Which, incidentally, is not the greatest API design but this is what we get with OpenAPI codegen. Though we do provide wrappers, I believe users can still import the generated things, so we should just major bump it.

**Review by jonathaningram — CHANGES_REQUESTED** [review]
Commented

**Comment by jkaho** [comment]
@jonathaningram ah got it, makes sense.

**Review by jonathaningram — APPROVED** [review]

**Review by jonathaningram — CHANGES_REQUESTED** [review]
Actually let's check this again re Account-Id header.

**Review by jonathaningram — APPROVED** [review]

**Pull request merged**